### ISSUE TITLE
`tff.learning.algorithms.build_weighted_fed_avg_with_optimizer_schedule` not working

### ISSUE THREAD

**Issue by shifdz** [tensorflow/federated#3072]
Hi, 

I am trying to train a federated learning model for a binary classification task in which I try to compare the performance between the standard federated averaging (FedAvg) and [ADAPTIVE FEDERATED OPTIMIZATION](https://arxiv.org/abs/2003.00295) (authored by @zcharles8).

My understanding is that `tff.learning.algorithms.build_weighted_fed_avg_with_optimizer_schedule` performs Adaptive federated optimization for which I have my optimizer function settings as follows:  I have Adam with a learning rate of 0.001 for client optimizer and (I tried with different learning rates) and SGD with a learning rate of 1.0 for the server optimizer (which I believe is the default). However, with this setting accuracy does not go beyond 0.5.

For standard federated averaging (FedAvg), I use `tff.learning.algorithms.build_weighted_fed_avg` with following settings:  Adam with a learning rate of 0.001 for client optimizer and and SGD with a learning rate of 1.0 for the server optimizer which works fine. 

My questions are as follows:

1. In standard federated averaging, does the client optimizer have to be 'SGD' only?

2. In the paper linked above it states "One such method is FEDAVG (McMahan et al., 2017), in which clients perform multiple epochs of SGD on their local datasets." Based on this statement, if the clients run Adam on their own loss function, it is not federated averaging?

3. What is the difference between ' standard federated averaging (FedAvg)' and 'Adaptive federated optimization (FedOpt) (paper linked above)' in terms of client and server optimizer functions?

In other words what is the different between `tff.learning.algorithms.build_weighted_fed_avg` and `tff.learning.algorithms.build_weighted_fed_avg_with_optimizer_schedule` ?

**Comment by zcharles8** [comment]
HI @shifdz. This kind of open-ended question is a better fit for a Stack Overflow question.

The distinction between "standard" FedAvg and FedOpt is discussed in the paper you mention. As for the difference between `tff.learning.algorithms.build_weighted_fed_avg` and `tff.learning.algorithms.build_weighted_fed_avg_with_optimizer_schedule`, this can be found in their docstrings. The latter is a generalization of the former where the client/server learning rates change over time.

I'm marking this as resolved for now. We encourage you to ask questions of this sort on Stack Overflow in the future.
